src/com/ableton/VirtualEnv.groovy

```diff
@@ -15,12 +15,59 @@ class VirtualEnv implements Serializable {
    * Destination directory for the virtualenv. This value is set during construction of
    * the object, and is under the system temporary directory.
    */
-  String destDir
+  String destDir = null
+  /**
+   * Series of commands needed to install and active a version of Python inside the
+   * current shell.
+   */
+  String envSetup = null
+
+  protected String activateSubDir = null
+
+  /**
+   * Create a virtualenv using a specific version of Python, installed via pyenv. pyenv
+   * should be installed on the node, but the actual setup of any required environment
+   * variables (e.g. PYENV_ROOT and PATH) will be done inside this function.
+   *
+   * @param script Script context.
+   *               <strong>Required value, may not be null!</strong>
+   * @param python Python version, as given by pyenv versions --list.
+   * @param pyenvRoot Path to the installation of pyenv.
+   * @return New instance of VirtualEnv object.
+   */
+  static VirtualEnv create(Object script, String python, String pyenvRoot) {
+    if (!script.isUnix()) {
```

Comment:
It's part of the Jenkins pipeline API, and thus part of the script context. `script` itself is not an `Object`, but it's technically a `WorkflowScript` object (which of course ultimately derives from `Object`). However, we can't call it a `WorkflowScript` because we don't have the library for that symbol in our project. Hence, we just type it as a generic `Object`.